assets/js/alerts.js

```diff
@@ -0,0 +1,236 @@
+
+    // update current filters
+    currentFilters[`${dropDownType}`] = dropDownText
+    // update alert table and filters
+    isReset = false
+    updateAll(alerts, currentFilters, isReset)
+  })
+
+  // on click of 'Reset Filters' button, clear filters, reset url params and update table and filters
+  $('#resetFilters').on('click', async e => {
+    e.preventDefault(e)
+    history.replaceState(null, '', '/alerts')
+    currentFilters = {
+      application: '',
+      environment: '',
+      namespace: '',
+      severity: '',
+    }
+    isReset = true
+    dropdownHandler.clearPendingValues()
+    updateAll(alerts, currentFilters, isReset)
+  })
+})
+
+const dropdownHandler = {
+  // values for unapplied dropdowns
+  pendingValues: {
+    application: '',
+    environment: '',
+    namespace: '',
+    severity: '',
+  },
+  clearPendingValues: function () {
+    this.pendingValues = {
+      application: '',
+      environment: '',
+      namespace: '',
+      severity: '',
+    }
+  },
+  updateDropdowns: function (filteredData, currentFilters, isReset) {
+    // Get dynamic options from filtered data
+    const applications = this.getOptions(filteredData, 'application')
+    const environments = this.getOptions(filteredData, 'environment')
+    const namespaces = this.getOptions(filteredData, 'namespace')
+    const severities = this.getOptions(filteredData, 'severity')
+
+    this.renderDropdown(applicationFilter, applications, currentFilters.application, 'application', isReset)
+    this.renderDropdown(environmentFilter, environments, currentFilters.environment, 'environment', isReset)
+    this.renderDropdown(namespaceFilter, namespaces, currentFilters.namespace, 'namespace', isReset)
+    this.renderDropdown(severityFilter, severities, currentFilters.severity, 'severity', isReset)
+  },
+  getOptions: function (data, key) {
+    const set = new Set(data.map(a => a.labels[`${key}`]))
+    return Array.from(set).sort()
+  },
+  removeOptions: function (selectElement) {
+    Array.from(selectElement.options).forEach(() => {
+      selectElement.remove(0)
+    })
+    const opt = document.createElement('option')
+    opt.value = ''
+    opt.textContent = ''
+    selectElement.appendChild(opt)
+  },
+  renderDropdown: function (select, options, selectedValue, key, isReset) {
+    if (isReset) {
+      this.pendingValues[key] = ''
+    }
+    selectedValue === select.value ? (this.pendingValues[key] = '') : (this.pendingValues[key] = select.value)
+    //  clear dropDown options to prevent duplicates
+    this.removeOptions(select)
+    // append options
+    options.forEach(option => {
+      const newOption = document.createElement('option')
+      newOption.value = option
+      newOption.textContent = option
+      selectedValue !== ''
+        ? (newOption.selected = option === selectedValue)
+        : (newOption.selected = option === this.pendingValues[key])
+      select.appendChild(newOption)
```

Comment:
Again, it's more understandable that this is assigning something if you do the value conditionally

```
newOption.selected = selectedValue !== '' ? option === selectedValue : option === this.pendingValues[key]
```